src/WebJobs.Script.WebHost/Security/Authentication/Keys/AuthenticationLevelHandler.cs

```diff
@@ -84,22 +83,25 @@ protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
 
         internal static Task<(string, AuthorizationLevel)> GetAuthorizationKeyInfoAsync(HttpRequest request, ISecretManagerProvider secretManagerProvider)
         {
-            // first see if a key value is specified via headers or query string (header takes precedence)
-            string keyValue = null;
-            if (request.Headers.TryGetValue(FunctionsKeyHeaderName, out StringValues values))
+            if (secretManagerProvider.SecretsEnabled)
```

Comment:
Can we reserve the if condition and return at the top if secrets is not enabled?

Reply:
Spoke with Lilian about this offline; reversing the condition would double the return statement, so keeping as-is.